examples/apps/resizable-inline/src/main.rs

```diff
@@ -0,0 +1,266 @@
+            }
+        }
+    });
+}
+
+fn run<B: Backend>(
+    terminal: &mut Terminal<B>,
+    app_state: &mut AppState,
+    rx: mpsc::Receiver<Event>,
+) -> Result<()>
+where
+    B::Error: Send + Sync + 'static,
+{
+    let mut redraw = true;
+    
+    loop {
+        if redraw {
+            terminal.draw(|frame| render(frame, app_state))?;
+        }
+        redraw = true;
+
+        match rx.recv()? {
+            Event::Input(key_event) => {
+                if key_event.kind == KeyEventKind::Press {
+                    match key_event.code {
+                        KeyCode::Char('q') => break,
+                        KeyCode::Char('+') | KeyCode::Char('=') => {
+                            app_state.increase_height();
+                            terminal.set_viewport_height(app_state.current_height)?;
+                        }
+                        KeyCode::Char('-') | KeyCode::Char('_') => {
+                            app_state.decrease_height();
+                            terminal.set_viewport_height(app_state.current_height)?;
+                        }
+                        KeyCode::Char('r') => {
+                            app_state.reset_height();
+                            terminal.set_viewport_height(app_state.current_height)?;
+                        }
+                        KeyCode::Char('i') => {
+                            app_state.insert_line();
+                            let count = app_state.insert_count;
+                            let now = app_state.start_time.elapsed();
+                            terminal.insert_before(1, |buf| {
+                                Paragraph::new(format!(
+                                    "📝 Inserted line #{} at {:02}:{:02}",
+                                    count,
+                                    now.as_secs() / 60,
+                                    now.as_secs() % 60
+                                ))
+                                .style(Style::default().fg(Color::Green).add_modifier(Modifier::BOLD))
+                                .render(buf.area, buf);
+                            })?;
+                        }
+                        _ => {}
+                    }
+                }
+            }
+            Event::Resize => {
+                terminal.autoresize()?;
+            }
+            Event::Tick => {
+                app_state.frame_count += 1;
+                redraw = false; // Don't redraw on every tick unless content changed
+            }
+        }
+    }
+    Ok(())
+}
+
+fn render(frame: &mut Frame, app_state: &AppState) {
+    let area = frame.area();
+    
+    // Create a block with border and title showing current height
+    let main_block = Block::default()
+        .borders(Borders::ALL)
+        .title(format!(" Resizable Viewport (Height: {}) ", app_state.current_height))
+        .title_style(Style::default().fg(Color::Cyan).add_modifier(Modifier::BOLD));
```

Comment:
nit: use stylize shortcuts and bordered() constructor:
```suggestion
    let main_block = Block::bordered()
        .title(format!(" Resizable Viewport (Height: {}) ", app_state.current_height).cyan().bold());
```

(and move the title on its own line if that needs to wrap)
```
let title = ...